packages/core/e2e/fixtures/test-plugins/list-query-plugin.ts

```diff
@@ -144,6 +144,12 @@ export class TestEntity extends VendureEntity implements Translatable, HasCustom
 
     @Column(() => TestEntityCustomFields)
     customFields: TestEntityCustomFields;
+
+    @ManyToOne(() => TestEntity, (type) => type.parent)
+    parent: TestEntity | null;
+
+    @Column('int', { nullable: true })
+    parentId: ID | null;
```

Comment:
Here I had to update the model a bit to repeat the error and take it into account in the tests

Reply:
above fix i think need consider more behavior
if we have `translations` like 
```ts
  @OneToMany(
    () => LensProcessOptionTranslation,
    (translation) => translation.base,
    { eager: true }
  )
  translations: Array<Translation<LensProcessOption>>;
```
the `parent` should be auto load `translations` otherwise it will result `translateEntity` failed

```ts

function translateLeaf(
    object: { [key: string]: any } | undefined,
    property: string,
    languageCode: LanguageCode | [LanguageCode, ...LanguageCode[]],
): any {
    if (object && object[property]) {
        if (Array.isArray(object[property])) {
            return object[property].map((nested2: any) => translateEntity(nested2, languageCode));
        } else if (object[property]) {
            return translateEntity(object[property], languageCode);
        }
    }
}

```
if `parent` existed but it have auto loaded `translations` it will broken here

Reply:
please see my latest updated repo https://github.com/tianyingchun/next-issue/tree/vendure_issue_2738

Reply:
```
InternalServerError [GraphQLError]: error.entity-has-no-translation-in-language
    at new I18nError (/Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/node_modules/@vendure/core/dist/i18n/i18n-error.js:21:9)
    at new InternalServerError (/Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/node_modules/@vendure/core/dist/common/error/errors.js:15:9)
    at translateEntity (/Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/node_modules/@vendure/core/dist/service/helpers/utils/translate-entity.js:33:15)
    at translateLeaf (/Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/node_modules/@vendure/core/dist/service/helpers/utils/translate-entity.js:104:20)
    at translateDeep (/Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/node_modules/@vendure/core/dist/service/helpers/utils/translate-entity.js:89:21)
    at TranslatorService.translate (/Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/node_modules/@vendure/core/dist/service/helpers/translator/translator.service.js:54:53)
    at file:///Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/packages/plugin-issue/src/services/menu.service.ts:82:76
    at Array.map (<anonymous>)
    at file:///Users/tianyingchun/Documents/coding/kzfoo/vendure-issue/packages/plugin-issue/src/services/menu.service.ts:82:37
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  path: undefined,
  locations: undefined,
  extensions: { code: 'INTERNAL_SERVER_ERROR' },
  variables: { entityName: 'Menu', languageCode: ',,en' },
  code: 'INTERNAL_SERVER_ERROR',
  logLevel: 0
}
```

Reply:
Interesting. I thought I had accounted for translations for all self-related relations. I will look into this and add join for eager links

> above fix i think need consider more behavior if we have `translations` like
> 
> ```ts
>   @OneToMany(
>     () => LensProcessOptionTranslation,
>     (translation) => translation.base,
>     { eager: true }
>   )
>   translations: Array<Translation<LensProcessOption>>;
> ```
> 
> the `parent` should be auto load `translations` otherwise it will result `translateEntity` failed
> 
> ```ts
> function translateLeaf(
>     object: { [key: string]: any } | undefined,
>     property: string,
>     languageCode: LanguageCode | [LanguageCode, ...LanguageCode[]],
> ): any {
>     if (object && object[property]) {
>         if (Array.isArray(object[property])) {
>             return object[property].map((nested2: any) => translateEntity(nested2, languageCode));
>         } else if (object[property]) {
>             return translateEntity(object[property], languageCode);
>         }
>     }
> }
> ```
> 
> if `parent` existed but it have auto loaded `translations` it will broken here

Reply:
seems to be a bug in the `translateDeep` function